### PR TITLE
[WIP] Add paraswap swap

### PR DESCRIPTION
**Pull request by Quazia**
Needs testing with https://github.com/rabbitholegg/questdk/pull/27

### CODE REVIEW

File: packages/paraswap/src/Paraswap.test.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+describe('Given the paraswap plugin', () => {
+  describe('When handling the bridge', () => {
+
+    test('should return a valid action filter', () => {
+      
+    })
+
+    test('should pass filter with valid transactions',  () => {
+      
+    })
```
Comment (jonathandiep) [inline packages/paraswap/src/Paraswap.test.ts:22]: Empty tests?

File: packages/paraswap/src/Paraswap.test.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    
+    test('should not pass filter with invalid transactions',  async () => {
```
Comment (jonathandiep) [inline packages/paraswap/src/Paraswap.test.ts:24]: I think the test case does pass? Seems like valid tx to me?

File: packages/paraswap/src/Paraswap.test.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      const filter = await swap({
+        chainId: ARB_ONE_CHAIN_ID,
+        contractAddress: AUGUSTUS_SWAPPER_ARBITRUM,
+        tokenIn: USDT_ADDRESS.toLowerCase() as Address,
+        tokenOut: USDCE_ADDRESS.toLowerCase() as Address,
+        amountIn: GreaterThanOrEqual(339000000),
+      })
+      expect(apply(transaction, filter)).to.be.true
+    })
+    test('should not pass filter with invalid transactions',  async () => {
```
Comment (jonathandiep) [inline packages/paraswap/src/Paraswap.test.ts:35]: I think the test case does pass? Seems like valid tx to me?

**Review by jonathandiep — APPROVED** [review]
Lets clean up the test file and I think we can go ahead and test out when ready

**Comment by Quazia** [comment]
@jonathandiep Ok - tests should be cleaned up now

File: packages/paraswap/src/Paraswap.test.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  describe('When handling the bridge', () => {
+    test('should return a valid action filter', async () => {
+      const filter = await swap({
+        chainId: ARB_ONE_CHAIN_ID,
+        contractAddress: AUGUSTUS_SWAPPER_ARBITRUM,
+        tokenIn: USDT_ADDRESS.toLowerCase() as Address,
+        tokenOut: USDCE_ADDRESS.toLowerCase() as Address,
+        amountIn: GreaterThanOrEqual(339000000),
+      })
+      console.log(filter.input)
```
Comment (jonathandiep) [inline packages/paraswap/src/Paraswap.test.ts:23]: [nit] Can remove this stray log

File: packages/paraswap/src/Paraswap.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        },
+      ],
+    },
+  })
+}
+
+export const getSupportedTokenAddresses = async (
+  _chainId: number,
+): Promise<Address[]> => {
+  // TODO: should we build this function for all chains? Might save a little compute having it pre-built once
```
Comment (jonathandiep) [inline packages/paraswap/src/Paraswap.ts:95]: I think would be nice to have a fallback static list of supported tokens

**Review by jonathandiep — APPROVED** [review]

**Pull request merged**